syft/frameworks/torch/tensors/add.py

```diff
@@ -0,0 +1,226 @@
+
+        for share in random_shares:
+            share.random_(mod)
+
+        shares = []
+        for i in range(n_workers):
+            if i == 0:
+                share = random_shares[i]
+            elif i < n_workers - 1:
+                share = random_shares[i] - random_shares[i - 1]
+            else:
+                share = secret - random_shares[i - 1]
+            shares.append(share)
+
+        return shares
+
+    @hook
+    def add(self, shares: dict, other_shares, *args, **kwargs):
+        """Adds two tensors together
+
+        Args:
+            shares: a dictionary <location_id -> PointerTensor) of shares corresponding to
+                self. Equivalent to calling self.child.
+            other_shares: a dictionary <location_id -> PointerTensor) of shares corresponding
+                to the tensor being added to self.
+        """
+
+        # if someone passes in a constant... (i.e., x + 3)
+        if not isinstance(other_shares, dict):
+            other_shares = torch.Tensor([other_shares]).share(*self.child.keys()).child
+
+        assert len(shares) == len(other_shares)
+
+        # matches each share which needs to be added according
+        # to the location of the share
+        new_shares = {}
+        for k, v in shares.items():
+            new_shares[k] = other_shares[k] + v
+
+        # return the true tensor (unwrapped - wrapping will happen
+        # automatically if needed)
+        response = AdditiveSharingTensor().set_shares(new_shares)
+
+        return response
+
+    @hook
+    def __add__(self, *args, **kwargs):
+        """Adds two tensors. See add() for more details."""
+        return self.add(*args, **kwargs)
+
+    @hook
+    def sub(self, shares: dict, other_shares, **kwargs):
+        """Subtracts an other tensor from self.
+
+        Args:
+            shares: a dictionary <location_id -> PointerTensor) of shares corresponding to
+                self. Equivalent to calling self.child.
+            other_shares: a dictionary <location_id -> PointerTensor) of shares corresponding
+                to the tensor being subtracted from self.
+        """
+
+        # if someone passes in a constant... (i.e., x - 3)
+        if not isinstance(other_shares, dict):
+            other_shares = torch.Tensor([other_shares]).share(*self.child.keys()).child
+
+        assert len(shares) == len(other_shares)
+
+        # matches each share which needs to be added according
+        # to the location of the share
+        new_shares = {}
+        for k, v in shares.items():
+            new_shares[k] = v - other_shares[k]
+
+        # return the true tensor (unwrapped - wrapping will happen
+        # automatically if needed)
+        response = AdditiveSharingTensor().set_shares(new_shares)
+
+        return response
+
+    @hook
+    def __sub__(self, *args, **kwargs):
+        """Subtracts two tensors. See .sub() forr details."""
+        return self.sub(*args, **kwargs)
+
```

Comment:
do we want mul?

Reply:
Yes - but not required to get federated learning working which is what i'm focused on atm